pkg/pulsar/pulsar.go

```diff
@@ -73,7 +73,7 @@ func NewClient(config Config) (*Client, error) {
 }
 
 func (c *Client) InitProducer() error {
-	_, err := c.consumer()
+	_, err := c.producer()
```

Comment:
This is a bug from the changes from #33, where the integration tests would have actually covered this problem :smile: